### PR TITLE
Add unit tests for transport procedure

### PR DESCRIPTION
**Pull request by Jax4Li**
### Describe what this PR does / why we need it
Add unit tests for transport procedure.
1. Client and Server transport procedure.
2. Observer mechanism
3. Data cache and version mechanism

### Does this pull request fix one issue?

Issues #13 

### Describe how you did it
Create fake some service implements in test cases, they can client.

### Special notes for reviews

### CODE REVIEW

**Comment by jnan806** [comment]
@Jax4Li   Could you please add some short comments at the `@Test` function ~

**Comment by Jax4Li** [comment]
> @Jax4Li Could you please add some short comments at the `@Test` function ~

sure

File: src/test/java/io/opensergo/OpenSergoClientTest.java [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+                    @Override
+                    public StreamObserver<SubscribeRequest> subscribeConfig(StreamObserver<SubscribeResponse> responseObserver) {
+                        return (FakerStreamObserver) subscribeRequest -> {
+                            if (StringUtils.isBlank(subscribeRequest.getResponseAck())) {
+                                actualRequest.set(subscribeRequest);
+                            }else {
+                                // from client ack, skip
+                                return;
+                            }
+                            // push first time data, it cloud be cached by client
```
Comment (jnan806) [inline src/test/java/io/opensergo/OpenSergoClientTest.java]: `cloud` is a typo of `could`

**Comment by jnan806** [comment]
The scope definition of `OpenSergoClientTest.java` is too large for your test cases, maybe `OpenSergoSubscribeConfigTest.java` or `OpenSergoUniversalTransportServiceTest.java` maybe is more suitable. 😃

**Comment by Jax4Li** [comment]
> The scope definition of `OpenSergoClientTest.java` is too large for your test cases, maybe `OpenSergoSubscribeConfigTest.java` or `OpenSergoUniversalTransportServiceTest.java` maybe is more suitable. 😃

Yeah, `OpenSergoUniversalTransportServiceTest.java` is ok,

**Review by jnan806 — APPROVED** [review]
LGTM， thanks !

**Comment by jnan806** [comment]
cc @sczyh30  any other questions ?

**Review by sczyh30 — APPROVED** [review]
LGTM

**Comment by sczyh30** [comment]
Nice work. Thanks!

**Pull request merged**